app/assets/javascripts/qpixel_api.js

```diff
@@ -151,23 +175,50 @@ window.QPixel = {
     $field.val(prev.substring(0, $field[0].selectionStart) + text + prev.substring($field[0].selectionEnd));
   },
 
+  /**
+   * Used to prevent launching multiple requests to /users/me
+   * @type {Promise<User>|null}
+   */
+  _pendingUser: null,
+
+  /**
+   * @type {User|null}
+   */
   _user: null,
 
   /**
    * Get the user object for the current user.
-   * @returns {Promise<Object>} a JSON object containing user details
+   * @returns {Promise<User>} a JSON object containing user details
    */
   user: async () => {
     if (QPixel._user != null || document.body.dataset.userId === 'none') {
       return QPixel._user;
     }
-    const resp = await fetch('/users/me', {
-      credentials: 'include',
-      headers: {
-        'Accept': 'application/json'
-      }
-    });
-    QPixel._user = await resp.json();
+
+    if(QPixel._pendingUser) {
+        return QPixel._pendingUser
+    }
+
+    try {
+        const myselfPromise = fetch('/users/me', {
+            credentials: 'include',
+            headers: {
+              'Accept': 'application/json'
+            }
+        });
+
+        QPixel._pendingUser = myselfPromise;
+
+        const resp = await myselfPromise;
+
+        QPixel._user = await resp.json();
+    } catch(error) {
+        throw error
```

Comment:
Hmm, it _technically_ can be done, yes, I was just trying to think ahead and rethrow because I have plans for proper error handling in such cases (unfortunately, currently, if we error during fetch, there is none). Another reason for doing a rethrow is just me being used to doing a `try..catch..finally` or `try..catch` — I've never done a `try..finally` before 😅. ECMAScript _does_ allow for `try..finally` statements (as one would expect), though, I'll drop the `catch` block!